pydicom/valuerep.py

```diff
@@ -1274,8 +1274,8 @@ def __init__(self, val: Union[str, float, Decimal],
         # If a string passed, then store it
         if isinstance(val, str):
             self.original_string = val.strip()
-        elif isinstance(val, (IS, ISfloat)) 
-                and hasattr(val, 'original_string'):
+        elif (isinstance(val, (IS, ISfloat)) 
```

Comment:
Now it complains about a trailing space... quite pedantic. For the future, we really may want to introduce auto-correct for these kinds of problems...